docs/content/docs/recipes/signals.md

```diff
@@ -0,0 +1,328 @@
+
+```typescript
+window.addEventListener("focus", () => {
+    this.sendFocusSignal(true);
+});
+window.addEventListener("blur", () => {
+    this.sendFocusSignal(false);
+});
+```
+```typescript
+private sendFocusSignal(hasFocus: boolean) {
+    this.signaler.submitSignal(
+        FocusTracker.focusSignalType,
+        { userId: this.audience.getMyself()?.userId, focus: hasFocus },
+    );
+}
+```
+To make sure all connected clients are notified of this focus change, we use the `onSignal` function to listen to the `focusSignalType`. We then update the local data using the payload information from the signal and emit a `focusChanged` event to re-render:
+
+```typescript
+this.signaler.onSignal(FocusTracker.focusSignalType, (clientId, local, payload) => {
+    this.onFocusSignalFn(clientId, payload);
+});
+```
+```typescript
+private readonly onFocusSignalFn = (clientId: string, payload: any) => {
+    const userId: string = payload.userId;
+    const hasFocus: boolean = payload.focus;
+
+    let clientIdMap = this.focusMap.get(userId);
+    if (clientIdMap === undefined) {
+        clientIdMap = new Map<string, boolean>();
+        this.focusMap.set(userId, clientIdMap);
+    }
+    clientIdMap.set(clientId, hasFocus);
+    this.emit("focusChanged");
+};
+```
+`FocusTracker` differs from `MouseTracker` as it uses the `Signal Request` pattern. This is when a newly joining client requests a specific signal to be sent from other connected clients, so that the new client can receive pertinent information immediately after connecting to the container.
+
+To achieve this pattern, `FocusTracker` defines the `focusRequestType` that will be sent to request the focus status of all the connected clients:
+
+```typescript
+private static readonly focusRequestType = "focusRequest";
+```
+`FocusTracker` then sends this signal request immediately after the new client connects to the container. Once again, the `onSignal` function is used to listen to this `focusRequestType` signal and each client responds to the signal request with their current focus status:
+
+```typescript
+this.signaler.onSignal(FocusTracker.focusRequestType, () => {
+    this.sendFocusSignal(document.hasFocus());
+});
+
+container.on("connected", () => {
+    this.signaler.submitSignal(FocusTracker.focusRequestType);
+});
+```
+
+## View
+
+In `app.ts`, we create instances of the `MouseTracker` and the `FocusTracker` to use in our application:
+
+```typescript
+/*...*/
+
+async function start(): Promise<void> {
+    /*...*/
+
+    // Render presence information for audience members
+    const contentDiv = document.getElementById("focus-content") as HTMLDivElement;
+    const mouseContentDiv = document.getElementById("mouse-position") as HTMLDivElement;
+    const focusTracker = new FocusTracker(
+        container,
+        services.audience,
+        container.initialObjects.signaler as Signaler,
```

Comment:
When the initialObjects was defined, the type was specified as "Signaler", so why do we need to cast it to Signaler?